pkg/promtail/targets/filetarget.go

```diff
@@ -206,6 +206,9 @@ func (t *FileTarget) sync() error {
 	toStopTailing := toStopTailing(matches, t.tails)
 	t.stopTailing(toStopTailing)
 
+	// Record the size of all the files being tailed.
+	t.updateTotalBytesMetric()
```

Comment:
I'd be more confident if we did this off `matches` - perhaps the bug is in the `missing`?